2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala

```diff
@@ -61,6 +61,12 @@ package model {
     enabled: Boolean,
     placeholder: Option[String]
   )
+  final case class RootResponseConfig(
+    enabled: Boolean = false,
+    statusCode: Int = 404,
+    headers: Map[String, String] = Map.empty[String, String],
+    body: String = ""
```

Comment:
on second thought, I don't think we need default values since config parsing will make sure everything is in order

Reply:
I may need some more context on desired behavior here. If functionality it truly optional, should a user need to put any configuration at all in the config file?

As it stands in this commit, the `rootResponse` key will need to be defined in the configuration file (when the value was an `Option`) it did not need to be added, and the config was backward compatible. 

If I start removing the default values, a user will need to define all of them in their configuration file even if they don't want to use this feature?

Current behavior if `rootResponse` block left out of hocon file:
```
java -jar ./target/scala-2.11/snowplow-stream-collector-0.12.0.jar --config ./examples/config.hocon
Exception in thread "main" pureconfig.error.ConfigReaderException: Cannot convert configuration to a com.snowplowanalytics.snowplow.collectors.scalastream.model.CollectorConfig. Failures are:
  at 'rootResponse':
    - (file:/Users/rbolkey/src/onespot/snowplow/2-collectors/scala-stream-collector/./examples/config.hocon:20) Key not found.
```

Behavior if I have an empty `rootResponse` block but remove the `headers` default value from `model.scala` and I don't define it in hocon file:

```
java -jar ./target/scala-2.11/snowplow-stream-collector-0.12.0.jar --config ./examples/config.hocon
Exception in thread "main" pureconfig.error.ConfigReaderException: Cannot convert configuration to a com.snowplowanalytics.snowplow.collectors.scalastream.model.CollectorConfig. Failures are:
  at 'rootResponse.headers':
    - (file:/Users/rbolkey/src/onespot/snowplow/2-collectors/scala-stream-collector/./examples/config.hocon:81) Key not found.
```

Reply:
As it stands, people do need to provide full configuration of the `cookieBounce`, `cookie`, `redirectMacro`, etc even if they don't want to use this feature, which is kind of a pain point.

Reply:
yes, so there are two levels to this issue:

- we've opted out of maintaining backwards compat for the config in #3556 so `RootResponseConfig` can't be an option itself
- inside `RootResponseConfig`, we can have options but in this case I don't think having options / default values for everything makes sense? maybe for body and headers but not for statusCode, wdyt?

I'm generally against options for the config as this might result in hidden behaviours.

Reply:
> As it stands, people do need to provide full configuration of the cookieBounce, cookie, redirectMacro, etc even if they don't want to use this feature, which is kind of a pain point.

I agree but it makes it harder to template as discussed in #3556

Reply:
> inside RootResponseConfig, we can have options but in this case I don't think having options / default values for everything makes sense? maybe for body and headers but not for statusCode, wdyt?

That makes sense to me. It's not surprising for free-form Strings and Collections to default to the empty value. The status code is enum like and doesn't have a sensible default. Closest was following the previous behavior.

Reply:
> I agree but it makes it harder to template as discussed in #3556

I looked at the thread there. I didn't see anyone propose using both Option at the top level along with an `enabled` field? You can kind of have both pieces of cake there. Missing block implies disabled (less pain for user's configuring). Present block has `enabled` defaulting to true, but you can template it to false if desired? Thoughts?

Reply:
I think it's a bit too ambiguous to have both.

Reply:
That's fine. I will attest that it was an annoying nuisance to have to deal with when configuring and upgrading the collector, but I'm just one data point :-).